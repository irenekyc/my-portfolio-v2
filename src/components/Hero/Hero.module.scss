.ikHero {
  &__banner {
    width: 100%;
    padding-top: 60px;
    &__inner {
      width: 100%;
      max-width: 600px;
      display: flex;
      -webkit-box-align: center;
      -ms-flex-align: center;
      align-items: center;
      text-align: center;
      margin-left: auto;
      margin-right: auto;
    }
    &__card {
      background: #303841;
      padding: 1.5rem;
      flex: 1;

      h3 {
        font-family: monospace;
        font-weight: 900;
        font-size: 3rem;
      }
    }
  }
}

// .hero-banner .hero-banner-inner #learning-hours {
//   -ms-grid-row: 2;
//   -ms-grid-column: 1;
//   grid-area: card1;
// }
// .hero-banner .hero-banner-inner #coding-hours {
//   -ms-grid-row: 1;
//   -ms-grid-column: 1;
//   -ms-grid-column-span: 2;
//   grid-area: card2;
//   -ms-flex-item-align: center;
//   -ms-grid-row-align: center;
//   align-self: center;
//   margin-right: auto;
//   margin-left: auto;
// }
// .hero-banner .hero-banner-inner #coding-hours h1 {
//   font-size: 3.3rem;
// }
// .hero-banner .hero-banner-inner #projects {
//   -ms-grid-row: 2;
//   -ms-grid-column: 2;
//   grid-area: card3;
// }
