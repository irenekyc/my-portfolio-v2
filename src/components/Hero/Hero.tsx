import React from "react";
import classnames from "classnames";
import styles from "./Hero.module.scss";

const Hero = () => {
  return (
    <section className={styles.ikHero__banner} id="hero" data-testid="hero">
      <div className={styles.ikHero__banner__inner}>
        <div
          className={classnames(
            styles.ikHero__banner__card,
            styles.ikHero__banner__card__learning
          )}
          id="learning-hours"
          data-testid="hero-card"
        >
          <h3 className="leading"> 300+</h3>
          <p> learning hours</p>
        </div>
        <div
          className={classnames(
            styles.ikHero__banner__card,
            styles.ikHero__banner__card__coding_hours
          )}
          id="coding-hours"
          data-testid="hero-card"
        >
          <h3 className="leading"> 5200+</h3>
          <p> coding hours</p>
        </div>
        <div
          className={classnames(
            styles.ikHero__banner__card,
            styles.ikHero__banner__card__projects
          )}
          id="projects"
          data-testid="hero-card"
        >
          <h3 className="leading"> 60+</h3>
          <p> coding projects</p>
        </div>
      </div>
    </section>
  );
};

export default Hero;
