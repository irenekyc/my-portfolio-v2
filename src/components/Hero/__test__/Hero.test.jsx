import React from "react";
import Hero from "../index";
import { render, screen } from "@testing-library/react";

it("Render Hero Section", () => {
  render(<Hero />);
  const hero = screen.getByTestId("hero");
  expect(hero).toBeInTheDocument();
  const heroCards = screen.getAllByTestId("hero-card");
  expect(heroCards).toHaveLength(3);
  expect(heroCards[0]).toHaveTextContent("learning hours");
  expect(heroCards[1]).toHaveTextContent("coding hours");
  expect(heroCards[2]).toHaveTextContent("coding projects");
});
