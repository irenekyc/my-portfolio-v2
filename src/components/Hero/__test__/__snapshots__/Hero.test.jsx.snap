// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Hero Section 1`] = `
<section
  class="ikHero__banner"
  data-testid="hero"
  id="hero"
>
  <div
    class="ikHero__banner__inner"
  >
    <div
      class="ikHero__banner__card ikHero__banner__card__learning"
      data-testid="hero-card"
      id="learning-hours"
    >
      <h3
        class="leading"
      >
         300+
      </h3>
      <p>
         learning hours
      </p>
    </div>
    <div
      class="ikHero__banner__card ikHero__banner__card__coding_hours"
      data-testid="hero-card"
      id="coding-hours"
    >
      <h3
        class="leading"
      >
         5200+
      </h3>
      <p>
         coding hours
      </p>
    </div>
    <div
      class="ikHero__banner__card ikHero__banner__card__projects"
      data-testid="hero-card"
      id="projects"
    >
      <h3
        class="leading"
      >
         60+
      </h3>
      <p>
         coding projects
      </p>
    </div>
  </div>
</section>
`;
